### PR TITLE
240719 maintenance

### PR DESCRIPTION
**Pull request by ogimenez-wirepas**
- Update to Python 3.12
- Add typing
- Add formatting with Black
- Update Github actions to check formatting and typing

### CODE REVIEW

File: wirepas_provisioning_server/data.py [hunk 1]
```diff
@@ -166,17 +166,17 @@ def __init__(self, config=None):
...
-        node_id=None,
-        node_role=None,
-        user_specific=None,
-        factory_key=None,
-    ):
+        uid: bytes,
+        method: int,
+        encryption_key: bytes,
+        authentication_key: bytes,
+        network_address: Optional[int],
```
Comment (GwendalRaoul) [inline wirepas_provisioning_server/data.py:170]: Is it really optional?

File: wirepas_provisioning_server/provisioning_server.py [hunk 2]
```diff
@@ -80,16 +81,25 @@ def loop(self, sleep_for=2):
...
+            dest: int,
+            src_ep: int,
+            dst_ep: int,
+            qos: int,
+            payload: bytes,
+        ) -> Optional[GatewayResultCode]:
         logging.debug("Sending packet (%s).", payload)
         try:
-            return self.interface.send_message(gw_id, sink_id, dest, src_ep, dst_ep, qos, payload)
+            return self.interface.send_message(gw_id, sink_id, dest, src_ep, dst_ep, payload, qos)
```
Comment (GwendalRaoul) [inline wirepas_provisioning_server/provisioning_server.py:92]: Is it a bug fix or was it changed also in the definition?

**Review by GwendalRaoul — APPROVED** [review]
Looks good to me.
Just two small comments to check

File: wirepas_provisioning_server/data.py (hunk 1 above)
Comment (ogimenez-wirepas) [reply wirepas_provisioning_server/data.py:170]: Yes, as per the [code](https://github.com/wirepas/wm-provisioning/blob/master/wirepas_provisioning_server/data.py#L112) and the [doc](https://github.com/wirepas/wm-provisioning/blob/master/examples/provisioning_config.yml#L22)

File: wirepas_provisioning_server/provisioning_server.py (hunk 2 above)
Comment (ogimenez-wirepas) [reply wirepas_provisioning_server/provisioning_server.py:92]: Surprisingly it is a bug fix, cf [doc](https://wirepas.github.io/wirepas-mqtt-library/wirepas_mqtt_library.html#wirepas_mqtt_library.wirepas_network_interface.WirepasNetworkInterface.send_message)
image

**Pull request merged**